unity-ggjj/Assets/Scripts/TextDecoder/NarrativeScript.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Ink.Runtime;
+using TextDecoder.Parser;
+using Object = Ink.Runtime.Object;
+using UnityEngine;
+
+[Serializable]
+public class NarrativeScript
+{
+    [field: Tooltip("Drag an Ink narrative script here.")]
+    [field: SerializeField] public TextAsset Script { get; private set; }
+
+    private ObjectStorage _objectStorage = new ObjectStorage();
+
+    public string Name => Script.name;
+    public Story Story { get; private set; }
+    public IObjectStorage ObjectStorage => _objectStorage;
+
+    /// <summary>
+    /// Initialise values on construction.
+    /// </summary>
+    /// <param name="script">An Ink narrative script</param>
+    /// <param name="actionDecoder">An optional action decoder, used for testing</param>
+    public NarrativeScript(TextAsset script, IActionDecoder actionDecoder = null)
+    {
+        Script = script;
+        Initialize(actionDecoder);
+    }
+
+    /// <summary>
+    /// Initialises script values that cannot be set in the Unity inspector
```

Comment:
Nit: Inconsistency: `Initialises` -> `Initializes`
```suggestion
    /// Initializes script values that cannot be set in the Unity inspector
```

Reply:
tfw british